packages/flutter_tools/lib/src/commands/build_aar.dart

```diff
@@ -155,6 +155,21 @@ class BuildAarCommand extends BuildSubCommand {
     if (remainingArguments.isEmpty) {
       return FlutterProject.current();
     }
-    return FlutterProject.fromDirectory(globals.fs.directory(findProjectRoot(globals.fs, remainingArguments.first)));
+    final File mainFile = globals.fs.file(remainingArguments.first);
+    final String path;
+    if (!mainFile.existsSync()) {
+      final Directory pathProject = globals.fs.directory(remainingArguments.first);
```

Comment:
Should we handle if the user passes the path to a file?

Reply:
This is handling both cases, file and directory

Reply:
won't `fs.directory(remainingArguments.first)` crash if passed the path to a file?

Reply:
It doesn't but this is only called after checking if mainFile doesn't exist. It first checks if the provided path is a File and then a directory if they both are not found on your file system it throws a tool exit

Reply:
Ahh I see, I forgot that constructing an invalid File or Directory is completely allowed. LGTM